src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -85,13 +91,67 @@ class DefaultSource
     new SequenceRowsRelation(config, sequenceId)(sqlContext)
   }
 
+  private def getFunctionResult(client: GenericClient[IO], functionId: Long, result: FunctionCall): IO[Json] = {
+    var res = result
+    while (res.status != None && res.status.get == "Running") {
+      res = client
+        .functionCalls(functionId)
+        .retrieveById(res.id.get)
+        .unsafeRunSync()
+    }
+
+    if (res.status != None && res.status.get == "Completed") {
+      for {
+        functionCallResponse <- client
+          .functionCalls(functionId)
+          .retrieveResponse(res.id.getOrElse(0L))
+      } yield functionCallResponse.response.getOrElse(Json.fromJsonObject(JsonObject.empty))
+    } else {
+      IO.pure(Json.fromJsonObject(JsonObject.empty))
+    }
+  }
+
+  private def callFunctionAndGetResult(config: RelationConfig, functionId: Long, data: Json): IO[Json] = {
+    implicit val backend: SttpBackend[IO, Nothing] = CdpConnector.retryingSttpBackend(config.maxRetries)
+    for {
+      client <- GenericClient.forAuth[IO](Constants.SparkDatasourceVersion, config.functionsAuth.get, config.baseUrl, apiVersion = Some("playground"))
+      call <- client.functionCalls(functionId).callFunction(Json.fromJsonObject(JsonObject.empty))
```

Comment:
Should be `callFunction(data)`?

Reply:
Yes, it is just that I have not found out what the input type should be. The Json has to be a Json object like `{ "data": <the data>}`. It would be nice to have input be Any, but I have not found any AnySerializers.